<a href="https://colab.research.google.com/github/Alviatul/293-TB-Alviatul-Nazila-Restoran/blob/master/Dokument_Search_Engine_232_293.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Anggota Kelompok :
1. Clarissa Sanindita Reikisyifa (202010370311232)
2. Alviatul Nazila               (202010370311293)

Kelas :Temu Kembali Informasi B

UTS : Dokument Search Engine, pencarian dokumen putusan narkotika menggunakal model Naive Bayes

# Import Library

In [ ]:
import pandas as pd
import numpy as np
! pip install kaggle
from sklearn.cluster import KMeans
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score, classification_report
from matplotlib import pyplot as plt
from sklearn.feature_selection import SelectKBest, f_classif
%matplotlib inline
plt.style.use('seaborn-whitegrid')
from sklearn.preprocessing import LabelEncoder, OneHotEncoder, OrdinalEncoder, StandardScaler, MinMaxScaler
import seaborn as sns


<ipython-input-60-9284ca02646f>:12: MatplotlibDeprecationWarning: The seaborn styles shipped by Matplotlib are deprecated since 3.6, as they no longer correspond to the styles shipped by seaborn. However, they will remain available as 'seaborn-v0_8-<style>'. Alternatively, directly use the seaborn API instead.
  plt.style.use('seaborn-whitegrid')


# Load Data

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
url = '/content/drive/MyDrive/overview.csv.csv'
df = pd.read_csv(url)
df.tail()

nomor lembaga_peradilan  \
45  630/Pid.Sus/2022/PN Srg         PN SERANG   
46  552/Pid.Sus/2023/PN SRG         PN SERANG   
47  558/Pid.Sus/2023/PN SRG         PN SERANG   
48  571/Pid.Sus/2023/PN SRG         PN SERANG   
49  615/Pid.Sus/2023/PN SRG         PN SERANG   

                                                bukti  
45  bukti berupa : 6 (enam) bungkus kecil plastic ...  
46  bukti berupa: 1 (satu) buah paket J&T dengan N...  
47  bukti berupa: 1 (satu) bungkus plastic bening ...  
48  bukti berupa: 1 (satu) bungkus plastic bening ...  
49  bukti berupa:- 1 (satu) unit handphone merk VI...

# Preprocessing

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 3 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   nomor              50 non-null     object
 1   lembaga_peradilan  50 non-null     object
 2   bukti              50 non-null     object
dtypes: object(3)
memory usage: 1.3+ KB


In [ ]:
df.isnull().sum()

nomor                0
lembaga_peradilan    0
bukti                0
dtype: int64

In [ ]:
df.duplicated().sum()

0

# Split Data Feature dan Target

In [ ]:
# Tambahkan kolom label terkait narkotika atau tidak
df['label'] = df['bukti'].str.contains('narkotika', case=False)

In [ ]:
# Vectorization
vectorizer = CountVectorizer()
X = vectorizer.fit_transform(df['bukti'])
y = df['label']

# Split Data Train dan Test

In [ ]:
# Pembagian Data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Model

In [ ]:
# Pelatihan Model Naive Bayes
nb_classifier = MultinomialNB()
nb_classifier.fit(X_train, y_train)


MultinomialNB()

# Evaluasi Model

In [ ]:
# Prediksi
y_pred = nb_classifier.predict(X_test)


In [ ]:
# Evaluasi Model
accuracy = accuracy_score(y_test, y_pred)
print(f'Accuracy: {accuracy:.2f}')

Accuracy: 0.90


In [ ]:
# Classification Report
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

       False       1.00      0.50      0.67         2
        True       0.89      1.00      0.94         8

    accuracy                           0.90        10
   macro avg       0.94      0.75      0.80        10
weighted avg       0.91      0.90      0.89        10



In [ ]:
df

nomor lembaga_peradilan  \
0   757/Pid.Sus/2022/PN Srg         PN SERANG   
1   647/Pid.Sus/2022/PN Srg         PN SERANG   
2   737/Pid.Sus/2022/PN Srg         PN SERANG   
3   746/Pid.Sus/2022/PN Srg         PN SERANG   
4   779/Pid.Sus/2022/PN Srg         PN SERANG   
5   812/Pid.Sus/2022/PN Srg         PN SERANG   
6   761/Pid.Sus/2022/PN Srg         PN SERANG   
7   646/Pid.Sus/2022/PN Srg         PN SERANG   
8   801/Pid.Sus/2022/PN Srg         PN SERANG   
9   738/Pid.Sus/2022/PN Srg         PN SERANG   
10  607/Pid.Sus/2022/PN Srg         PN SERANG   
11  792/Pid.Sus/2022/PN Srg         PN SERANG   
12  606/Pid.Sus/2022/PN Srg         PN SERANG   
13  745/Pid.Sus/2022/PN Srg         PN SERANG   
14  857/Pid.Sus/2022/PN Srg         PN SERANG   
15  565/Pid.Sus/2022/PN Srg         PN SERANG   
16  731/Pid.Sus/2022/PN Srg         PN SERANG   
17  782/Pid.Sus/2022/PN Srg         PN SERANG   
18  783/Pid.Sus/2022/PN Srg         PN SERANG   
19  786/Pid.Sus/2022/PN Srg         PN SERANG   
20  657/Pid.Sus/2022/PN Srg         PN SERANG   
21  700/Pid.Sus/2022/PN Srg         PN SERANG   
22  693/Pid.Sus/2022/PN Srg         PN SERANG   
23  856/Pid.Sus/2022/PN Srg         PN SERANG   
24  707/Pid.Sus/2022/PN Srg         PN SERANG   
25  655/Pid.Sus/2022/PN Srg         PN SERANG   
26  785/Pid.Sus/2022/PN Srg         PN SERANG   
27  695/Pid.Sus/2022/PN Srg         PN SERANG   
28  694/Pid.Sus/2022/PN Srg         PN SERANG   
29  591/Pid.Sus/2022/PN Srg         PN SERANG   
30  681/Pid.Sus/2022/PN Srg         PN SERANG   
31  689/Pid.Sus/2022/PN Srg         PN SERANG   
32  656/Pid.Sus/2022/PN Srg         PN SERANG   
33  756/Pid.Sus/2022/PN Srg         PN SERANG   
34  686/Pid.Sus/2022/PN Srg         PN SERANG   
35  648/Pid.Sus/2022/PN Srg         PN SERANG   
36  690/Pid.Sus/2022/PN Srg         PN SERANG   
37  680/Pid.Sus/2022/PN Srg         PN SERANG   
38  748/Pid.Sus/2022/PN Srg         PN SERANG   
39  618/Pid.Sus/2022/PN Srg         PN SERANG   
40  791/Pid.Sus/2022/PN Srg         PN SERANG   
41  684/Pid.Sus/2022/PN Srg         PN SERANG   
42  691/Pid.Sus/2022/PN Srg         PN SERANG   
43  572/Pid.Sus/2022/PN Srg         PN SERANG   
44  683/Pid.Sus/2022/PN Srg         PN SERANG   
45  630/Pid.Sus/2022/PN Srg         PN SERANG   
46  552/Pid.Sus/2023/PN SRG         PN SERANG   
47  558/Pid.Sus/2023/PN SRG         PN SERANG   
48  571/Pid.Sus/2023/PN SRG         PN SERANG   
49  615/Pid.Sus/2023/PN SRG         PN SERANG   

                                                bukti  label  
0   bukti berupa : 1 (satu) bungkus plastik bening...   True  
1   bukti berupa : narkotika berat netto akhir 0,0...   True  
2   bukti berupa : - 1 (satu) bungkus plastik beni...   True  
3   bukti berupa : 3 (tiga) bungkus plastic bening...   True  
4   bukti berupa: 1 (satu) bungkus plastik klip be...   True  
5   bukti berupa : narkotika berat netto akhir 0,0...   True  
6   bukti berupa: 1 (satu) paket plastik bening be...   True  
7   bukti berupa: 1 (satu) unit Handphone merk Viv...  False  
8   bukti berupa : (empat)bungkus plastic bening y...   True  
9   bukti berupa : 1 (satu) bungkus plastik bening...   True  
10  bukti berupa: 1 (satu) buku tabungan Tahapan B...  False  
11  bukti berupa: - 3 (dua) bungkus plastik klip b...   True  
12  bukti berupa: 2 (dua) bungkus besar plastic be...   True  
13  bukti berupa: 1 (satu) bungkus plastik bening ...   True  
14  bukti berupa: - 1 (satu) buah handphone merk O...  False  
15  bukti berupa : 21 bungkus pelastik bening beri...   True  
16  bukti berupa: 20 (dua puluh) bungkus plastik b...   True  
17  bukti berupa: 1 (satu) buah Simcard XL dengan ...  False  
18  bukti berupa 1 (satu) unit telephone genggam m...  False  
19  bukti berupa :1 (satu) bungkus narkotika jenis...   True  
20  bukti berupa: 1 (satu) paket lakban hitam yang...   True  
21  bukti berupa : 1 (satu) buah plastic warna hit...   True  
22  bukti berupa: 3 (tiga) bungkus plastik klip be...   True  
23  bukti 

# Kesimpulan

Berdasarkan hasil evaluasi model Naive Bayes, yang digunakan untuk klasifikasi dokumen putusan narkotika, model tersebut menunjukkan kinerja yang cukup baik dengan tingkat akurasi keseluruhan sebesar 90%. Kinerja ini tercermin dalam nilai presisi yang tinggi (0.94 pada rata-rata makro dan 0.91 pada rata-rata tertimbang), yang menunjukkan keandalan model dalam mengidentifikasi label yang benar. Namun, perlu diperhatikan bahwa meskipun model memiliki recall sempurna (1.00) untuk kategori 'True', terdapat penurunan signifikan pada recall untuk kategori 'False' (hanya 0.50). Hal ini menunjukkan bahwa model cenderung lebih akurat dalam mengidentifikasi kasus positif daripada negatif. Skor F1, yang merupakan rata-rata harmonis antara presisi dan recall, juga mendukung kesimpulan ini dengan nilai yang lebih tinggi untuk kategori 'True' (0.94) dibandingkan dengan 'False' (0.67). Kesimpulannya, model ini efektif dalam mengidentifikasi dokumen yang relevan, namun ada ruang untuk peningkatan dalam keseimbangan antara identifikasi positif dan negatif.